brozzler/worker.py

```diff
@@ -480,34 +482,39 @@ def _on_service_worker_version_updated(chrome_msg):
         return outlinks
 
     def _fetch_url(self, site, url=None, page=None):
-        proxies = None
+        proxy_url = self._proxy_for(site)
+
         if page:
             url = page.url
-        if self._proxy_for(site):
-            proxies = {
-                "http": "http://%s" % self._proxy_for(site),
-                "https": "http://%s" % self._proxy_for(site),
-            }
+
+        if proxy_url:
+            http = urllib3.ProxyManager("http://%s" % proxy_url, cert_reqs="CERT_NONE")
+        else:
+            http = urllib3.PoolManager()
+
         user_agent = site.get("user_agent")
         headers = {"User-Agent": user_agent} if user_agent else {}
         headers.update(site.extra_headers(page))
 
         self.logger.info("fetching url %s", url)
         try:
             # response is ignored
-            requests.get(
+            http.request(
```

Comment:
It would be nice if we had an equivalent of max fetch time, but I don't see anything available out of the box, so it can wait.